build-system/cpp-cmake-conan/build.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2024 Contributors to the Eclipse Foundation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import subprocess
+from argparse import ArgumentParser
+
+from velocitas_lib import get_workspace_dir
+
+CMAKE_EXECUTABLE = "cmake"
+CONAN_EXECUTABLE = "conan"
+
+
+def safe_get_workspace_dir() -> str:
+    """A safe version of get_workspace_dir which defaults to '.'."""
+    try:
+        return get_workspace_dir()
+    except Exception:
+        return "."
+
+
+def print_build_info(
+    build_variant: str, build_arch: str, build_target: str, is_static_build: bool
+) -> None:
+    """Print information about the build.
+
+    Args:
+        build_variant (str): The variant of the build: "release" or "debug"
+        build_arch (str): The architecture the app is built for.
+        build_target (str): Which artefact is being built.
+        is_static_build (bool): Enable static building.
+    """
+    cmake_version = subprocess.check_output(
+        [CMAKE_EXECUTABLE, "--version"], encoding="utf-8"
+    ).strip()
+    conan_version = subprocess.check_output(
+        [CONAN_EXECUTABLE, "--version"], encoding="utf-8"
+    ).strip()
+
+    print(f"CMake version      {cmake_version}")
+    print(f"Conan version      {conan_version}")
+    print(f"Build arch         {build_arch}")
+    print(f"Build variant      {build_variant}")
+    print(f"Build target       {build_target}")
+    print(f"Static build       {'yes' if is_static_build else 'no'}")
+
+
+def build(
+    build_variant: str, build_arch: str, build_target: str, static_build: bool
+) -> None:
+    CMAKE_CXX_FLAGS = "--coverage -g -O0"
+    build_folder = os.path.join(safe_get_workspace_dir(), "build")
+    if build_variant == "release":
+        CMAKE_CXX_FLAGS = "--coverage -s -g -O3"
+
+    os.makedirs(build_folder, exist_ok=True)
+
+    # Expose the PATH of the build-time requirements from Conan to CMake - this is NOT handled by
+    # any of Conan's CMake generators at the moment, hence we parse the conanbuildinfo.txt which
+    # is generated and holds these paths. This allows us to always use the protoc and grpc cpp plugin
+    # of the build system.
+    BUILD_TOOLS_PATH = ""
```

Comment:
Yes because in Conan 1 with cmaketool integration, build machine tools are not on the path, e.g. protoc and grpc_cpp_gen. It will fail to build proto files when cross compiling, hence why we need the BUILD_TOOLS_PATH.

Reply:
but the path is empty and never gets filled?

Reply:
accidentially pushed an outdated version